File: Week_01/id_86/LeetCode_083_086.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AlgorithmTest
{ 
    public class LeetCode_083_086
    {
        /// <summary>
        /// 92ms 23.9MB
        /// </summary>
        /// <param name="head"></param>
        /// <returns></returns>
        public ListNode MethodFor093_1(ListNode head)
        {
            if (head == null)
                return null;
            if (head.next == null)
                return head;
            else
            {
                ListNode listNode = head;
                while (listNode.next != null)
                {
                    if (listNode.next.val == listNode.val)
                    {
                        listNode.next = listNode.next.next;
                    }
                    else
                    {
                        listNode = listNode.next;
                    }
                }
            }

            return head;
        }
    }
}

File: Week_01/id_86/LeetCode_083_086.c
//simple C

void MethodFor083_1(struct ListNode* head)
{
	if (head == NULL)
	{
		return NULL;
	}
	struct ListNode *p, *q;
	p = head;
	while (p->next)
	{
		q = p->next;
		if (p->val == q->val)
		{
			p->next = q->next;
		}
		else
		{
			p = q;
		}
	}
	return head;
}